### PR TITLE
 Enable the vscode tests to work

### LINKED ISSUE #324: VS Code integration tests is failing.
Now the tests are failing by throwing the following error:

HTTPError: Response code 404 (Not Found)
    at Request._onResponseBase (/Users/runner/work/liberty-tools-vscode/liberty-tools-vscode/node_modules/got/dist/source/core/index.js:913:31)
    at Request._onResponse (/Users/runner/work/liberty-tools-vscode/liberty-tools-vscode/node_modules/got/dist/source/core/index.js:948:24)
    at ClientRequest.<anonymous> (/Users/runner/work/liberty-tools-vscode/liberty-tools-vscode/node_modules/got/dist/source/core/index.js:962:23)
    at Object.onceWrapper (node:events:633:26)
    at ClientRequest.emit (node:events:530:35)

### PR DESCRIPTION
**Pull request by vaisakhkannan**
Fixes #324 Enabled the vscode tests to work, enabled the clear command history at the end of gradle project tests to avoid receiving 'recently used' suggestions, and updated the version of the clipboardy component from 4.0.0 to 2.3.0.

### CODE REVIEW

File: src/test/GradleTestDevModeActions.ts [hunk 1]
```diff
@@ -263,6 +266,21 @@ it('View test report for gradle project', async () => {
 }).timeout(30000);
 
 
+  // Now, clear the command history of the "command palette" to avoid receiving "recently used" suggestions. This action should be performed at the end of Gradle Project tests.
+it('Clear Command Palatte', async () => {
+  await new Workbench().executeCommand('Clear Command History');
```
Comment (aparnamichael) [inline src/test/GradleTestDevModeActions.ts:271]: can we move this as a function in testUtils class so that we can reuse this whenever we want.

Comment (vaisakhkannan) [reply src/test/GradleTestDevModeActions.ts]: Yes, moved to testUtils class

File: src/test/utils/testUtils.ts [hunk 2]
```diff
@@ -179,4 +179,19 @@ export async function stopLibertyserver() {
...
+export async function clearCommandPalette() {
+  await new Workbench().executeCommand('Clear Command History');
+  await delay(30000);  
+  const dialog = new ModalDialog();
+  const message = await dialog.getMessage();
+
+  expect(message).contains('Do you want to clear the history of recently used commands?');
+  const details = await dialog.getDetails();
+
+  expect(details).equals(`This action is irreversible!`);
```
Comment (aparnamichael) [inline src/test/utils/testUtils.ts:192]: No need to check for details. message title and button check is enough.

Comment (vaisakhkannan) [reply src/test/utils/testUtils.ts]: Removed the check for details of dialog

**Review by aparnamichael — APPROVED** [review]
Looks good to me

File: src/test/GradleTestDevModeActions.ts [hunk 3]
```diff
@@ -263,6 +264,10 @@ it('View test report for gradle project', async () => {
 }).timeout(30000);
 
 
+  // Now, clear the command history of the "command palette" to avoid receiving "recently used" suggestions. This action should be performed at the end of Gradle Project tests.
```
Comment (TrevCraw) [inline src/test/GradleTestDevModeActions.ts:267]: what happens if we don't clear the history? We should explain that in the comment here.

File: package.json [hunk 4]
```diff
@@ -247,6 +247,7 @@
 		"@types/glob": "^7.1.1",
 		"@types/mocha": "^7.0.2",
 		"@types/node": "^16.11.7",
+		"@types/selenium-webdriver": "^4.1.22",
```
Comment (TrevCraw) [inline package.json:250]: what are the `@types/selenium-webdriver` and `selenium-webdriver` dependencies needed for?

**Review by TrevCraw — CHANGES_REQUESTED** [review]

File: package.json (hunk 4 above)
Comment (vaisakhkannan) [reply package.json:250]: `selenium-webdriver` automates browser tasks like clicking buttons, filling out forms, navigating between pages, and more.
`@types/selenium-webdriver` enhances code editor features like auto-completion, type-checking, and inline documentation.

I removed both and tried running the build, but it failed with the following error.

```
ERROR in /liberty-tools-vscode/src/test/GradleInitProject.ts(2,11)
      TS2305: Module '"vscode-extension-tester"' has no exported member 'WebDriver'.

ERROR in liberty-tools-vscode/src/test/utils/macUtils.ts(12,16)
      TS2339: Property 'click' does not exist on type 'DefaultTreeItem'.

ERROR in /liberty-tools-vscode/src/test/utils/macUtils.ts(22,21)
      TS2339: Property 'click' does not exist on type 'InputBox'.

ERROR in /liberty-tools-vscode/src/test/utils/testUtils.ts(46,17)
      TS2339: Property 'click' does not exist on type 'InputBox'.

ERROR in /liberty-tools-vscode/src/test/utils/testUtils.ts(138,23)
      TS2339: Property 'getDriver' does not exist on type 'Workbench'.

ERROR in /liberty-tools-vscode/src/test/utils/testUtils.ts(161,23)
      TS2339: Property 'getDriver' does not exist on type 'Workbench'.

ERROR in /liberty-tools-vscode/src/test/utils/testUtils.ts(179,17)
      TS2339: Property 'click' does not exist on type 'InputBox'.

8 errors have detailed information that is not shown.
Use 'stats.errorDetails: true' resp. '--stats-error-details' to show it.

webpack 5.85.0 compiled with 8 errors and 2 warnings in 1345 ms


```

I then added @types/selenium-webdriver: ^4.1.22 back to the devDependencies, and now the build starts running.

File: package.json [hunk 5]
```diff
@@ -257,22 +258,26 @@
...
 		"gradle-to-js": "^2.0.0",
-		"@types/semver": "^7.3.2",
+		"jsonpath-plus": "^7.0.0",
+		"node-fetch": "^2.6.1",
+		"path": "^0.12.7",
+		"selenium-webdriver": "^4.0.0",
 		"vscode-languageclient": "^8.1.0",
-		"xml2js": "^0.5.0",
-		"jsonpath-plus": "^7.0.0"
+		"xml2js": "^0.5.0"
```
[the remark this answers is not in the document]
Comment (vaisakhkannan) [reply package.json:277]: I removed the `node-fetch` and `path` dependencies, and added the `clipboardy` and `selenium-webdriver `dependencies only to the `devDependencies` section.

File: src/test/GradleTestDevModeActions.ts [hunk 6]
```diff
@@ -8,8 +8,7 @@ describe('Devmode action tests for Gradle Project', () => {
     let sidebar: SideBarView;
     let debugView: DebugView;
     let section: ViewSection;
-    let item: DefaultTreeItem;
-    let menu: ViewItem[];
```
[the remark this answers is not in the document]
Comment (vaisakhkannan) [reply src/test/GradleTestDevModeActions.ts:12]: I have removed it

File: src/test/GradleTestDevModeActions.ts [hunk 7]
```diff
@@ -29,8 +28,10 @@ it('getViewControl works with the correct label',  async() => {
 
 it('Open dasboard shows items - Gradle', async () => {
 
-    
-    await utils.delay(80000);    
+      
+    await utils.delay(65000);
+    await section.expand(); 
+    await utils.delay(6000);
```
[the remark this answers is not in the document]
Comment (vaisakhkannan) [reply src/test/GradleTestDevModeActions.ts:34]: done

File: src/test/MavenTestDevModeActions.ts [hunk 8]
```diff
@@ -29,9 +29,11 @@ it('getViewControl works with the correct label',  async() => {
...
 it('Open dasboard shows items - Maven', async () => {
 
-    
-  await utils.delay(65000);    
-  const menu = await section.getVisibleItems();            
+  
+  await utils.delay(65000);
+  section.expand();
+  await utils.delay(6000);
+  const menu = await section.getVisibleItems();
```
[the remark this answers is not in the document]
Comment (vaisakhkannan) [reply src/test/MavenTestDevModeActions.ts:36]: Done @TrevCraw

**Review by TrevCraw — APPROVED** [review]

**Review by aparnamichael — APPROVED** [review]

**Pull request merged**